config/secure-headers.php

```diff
@@ -316,47 +316,227 @@
 	/*
 	 * There is no easy way to use CSP with debug bar at the moment so we disable CSP if debug bar is enabled.
 	 */
-	'custom-csp' => ((bool) env('DEBUGBAR_ENABLED', false)) ? '' : null,
-
 	'csp' => [
+		'enable' => ((bool) env('DEBUGBAR_ENABLED', false)) === false,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
 		'report-only' => false,
-		'report-uri' => null,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to
+		'report-to' => '',
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri
+		'report-uri' => [
+			// uri
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content
 		'block-all-mixed-content' => false,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests
 		'upgrade-insecure-requests' => false,
 
-		/*
-		 * Please references script-src directive for available values, only `script-src` and `style-src`
-		 * supports `add-generated-nonce`.
-		 *
-		 * Note: when directive value is empty, it will use `none` for that directive.
-		 */
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/base-uri
+		'base-uri' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/child-src
+		'child-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src
+		'connect-src' => [
+			'https://lycheeorg.github.io/update.json',
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+		'default-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src
+		'font-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action
+		'form-action' => [
+			'self' => true,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors
+		'frame-ancestors' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src
+		'frame-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+		'img-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/manifest-src
+		'manifest-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src
+		'media-src' => [
+			'self' => true,
+			'allow' => [
+				'blob:',
+			],
```

Comment:
```suggestion
			'allow' => [],
```

I hope this removes the annoying empty `blob:` element from the the header. Currently it is

```
default-src 'self' blob:;
```

The change should give us

```
default-src 'self';
```

Reply:
Removed in [3f5583f](https://github.com/LycheeOrg/Lychee/pull/1528/commits/3f5583f3f4300551dbb90229ed1d7e908202272d)